File: solutions/c/hamming/1/hamming.c
#include "hamming.h"
#include <string.h>

int compute(const char *lhs, const char *rhs)
{
    long unsigned int rlen = strlen(rhs);
    if (strlen(lhs) != rlen) return -1;
    int d = 0;
    for (long unsigned int i = 0; i < rlen; i++)
    {
        if (lhs[i] - rhs[i] != 0) d++;
    }
    return d;
}

File: solutions/ruby/hamming/1/hamming.rb
class Hamming
  def self.compute(str1, str2)
    raise ArgumentError if str1.length != str2.length
    distance = 0
    for i in 0...str1.length
      distance += 1 if str1[i] != str2[i]
    end
    distance
  end
end